include/zephyr/net/socket.h

```diff
@@ -945,8 +946,18 @@ struct net_socket_register {
 
 #define NET_SOCKET_REGISTER(socket_name, prio, _family, _is_supported, _handler) \
 	static const STRUCT_SECTION_ITERABLE(net_socket_register,	\
-			NET_SOCKET_GET_NAME(socket_name, prio)) = {		\
+			NET_SOCKET_GET_NAME(socket_name, prio)) = {	\
 		.family = _family,					\
+		.is_offloaded = false,					\
+		.is_supported = _is_supported,				\
+		.handler = _handler,					\
+	}
+
+#define NET_SOCKET_OFFLOAD_REGISTER(socket_name, prio, _family, _is_supported, _handler) \
+	static const STRUCT_SECTION_ITERABLE(net_socket_register,	\
+			NET_SOCKET_GET_NAME(socket_name, prio)) = {	\
+		.family = _family,					\
+		.is_offloaded = true,					\
 		.is_supported = _is_supported,				\
 		.handler = _handler,					\
 	}
```

Comment:
As these macros are basically doing the same thing, I was pondering if we could simplify this a bit, so something like this
```
#define _NET_SOCKET_REGISTER(socket_name, prio, _family, _is_supported, _handler, _is_offloaded)  \
...

#define NET_SOCKET_REGISTER(socket_name, prio, _family, _is_supported, _handler) \
	_NET_SOCKET_REGISTER(socket_name, prio, _family, _is_supported, _handler, false)

#define NET_SOCKET_OFFLOAD_REGISTER(socket_name, prio, _family, _is_supported, _handler) \
	_NET_SOCKET_REGISTER(socket_name, prio, _family, _is_supported, _handler, true)
```

Reply:
Good point, I've applied your suggestion.